### ISSUE TITLE
Windows 8.1 and 10 binaries needed

### ISSUE THREAD

**Issue by sv1uy** [maqifrnswa/PCSI#6]
Hi,

I just installed psciGUI in Windows 8.1 and Windows 10 PCS and when I try to load an image, the program closes!
I even tried a 258,650 byte PNG and JPG image!
Still closes!

73 de Demetre M0SUY/SV1UY

**Comment by maqifrnswa** [comment]
Thanks - I'll look in to it. I've used it on Windows 7 and Linux (Ubuntu 20.04), so I haven't tried on Windows 8.1 or 10.

If you have a chance, could you try running "pcsiGUI.exe" from a windows command line or powershell? When it crashes, could you please post here with what it says in the command line/powershell window?

**Comment by sv1uy** [comment]
Hi,

Thanks for quick reply. I did run it from a powershell and started pcsiGUI.exe using ./ infront, but when I attempted to load a file, the program just closed and nothing was reported in the powershell.

73 de Demetre M0SUY/SV1UY

**Comment by maqifrnswa** [comment]
Thanks again for checking that out. I borrowed a windows 10 laptop and confirmed this bug.  I updated the release text to say that they are only for windows 7 for now. I'll try to set up a build environment for windows 10 and can give updated files to test later.

**Comment by sv1uy** [comment]
Great Scott,

Thanks for letting me know! I can't wait to try the new mode!
SSTV is one of my favourite modes and I trust PCSI will be one of them too!

---
73 de Demetre - M0SUY/SV1UY
e-mail: demetre.sv1uy@gmail.com
WINLINK Radio e-mail: m0suy@winlink.org
WEB: http://www.qsl.net/sv1uy
APRS: Best Social Network for Radio Amateurs

**Comment by maqifrnswa** [comment]
Long story short: it works on Windows 10 if you follow the python build instructions, but the compiled binaries don't work. The installer software that takes the working python program and packages it into an executable binary is screwing up somewhere. I have a windows 10 laptop now, so I can keep trying things out. I may have to abandon using msys and figure everything out using MSVC (which I'm not that familiar with). If there are any Windows developers out there that can help up turning the working python program into an executable file, I could appreciate the help! It's frustrating because I see it working and am using it as a python program, but when it gets turned in to a binary executable, it stops.

It's related to the PIL tkinter glue, there's a segfault and windows access violation when loading images if you use the .exe file that does not happen if you run the .py file.  I've tried pyinstaller and cx_freeze to convert the .py to an .exe., neither are working. It may be that they are designed for MSVC, not msys/mingw.

So if you'd like to try it out on Windows 10 in the mean time, checkout README.md for (complicated) instructions on how to get it working. Windows 7 and Linux users can still just download the binaries.

here's the error:
```
Windows fatal exception: access violation

Current thread 0x00002818 (most recent call first):
  File "site-packages/PIL/ImageTk.py", line 203 in paste
  File "site-packages/PIL/ImageTk.py", line 121 in __init__
  File "pcsiGUI.py", line 148 in loadfile
  File "tkinter/__init__.py", line 1883 in __call__
  File "tkinter/__init__.py", line 1420 in mainloop
  File "pcsiGUI.py", line 443 in <module>
```

**Comment by maqifrnswa** [comment]
I built it entirely using MSVC, and the image loading bug is gone. If you get a sec, could you try: https://github.com/maqifrnswa/PCSI/releases/download/v0.0.0/pcsiGUI-windows10-64.zip
I had to fight with getting liblbfgs built properly, so the "Process PCSI" button may not work. It should work, but I haven't been able to test it.

I very much appreciate the debugging, especially since I don't have Macs or Windows 10 development systems easily available!

**Comment by sv1uy** [comment]
Hi,

Thanks for the new pcsiGUI. I tested it and it works in both Windows
10-64bit and Windows 8.1-64bit I have here!
Tomorrow I will attempt to try it via APRS here in my house with 2 VHF
Radios!

I've read in the APRSSIG mailing list that Bob is very fond of PCSI but he
suggest we don't use it on the VHF APRS Frequency.
He proposes we set an object on the APRS frequency indicating the PCSI QRG!

I hope I can transfer pictures with it here at home first and then try to
have PCSI QSOS with others!

I am QRV everyday on 30m ROBUST PACKET and PACKET APRS on 10147.30 kHZ USB
DIAL and I would love to start using it
there too, although propagation on HF is bad!

I will let you know of the outcome!

---
73 de Demetre - M0SUY/SV1UY
e-mail: demetre.sv1uy@gmail.com
WINLINK Radio e-mail: m0suy@winlink.org
WEB: http://www.qsl.net/sv1uy
APRS: Best Social Network for Radio Amateurs

**Comment by PA3GJX** [comment]
Screenshot from my /MM Windows 10 Pro version 1909 Build 18363.900 

Thanks4all work done. 
![IMG_20200706_160441](https://user-images.githubusercontent.com/67911667/86602881-95155880-bfa3-11ea-8c82-4dee53584e7b.jpg)

**Comment by maqifrnswa** [comment]
@PA3GJX that's great info, thanks. it looks like it was running fine and crashed when you tried to start image transmission. Could you let me know what the screen shot looked like before you transmitted? A setting might be off. For example, if the image ID >255, this could happen. I can put in some data validation to prevent that and warn users. Thanks for testing it out! 73

**Comment by PA3GJX** [comment]
Hello Scott,

Indeed the picture was too big, about 2,1 Mb. But is assumed because it was
Cropped it could be transmitted. ;-)

When today i tried to load a smaller 50kb .jpg in the other screen i could
see data.

It is also nice to have the warning there is no TNC connected. I use a SCS
ptc3 with robust packet. Have to find a other station rx/tx though.

Question.

On my Windows 10 the gui does not save the station information, can it be i
have done something wrong installing?

Thanks for this new picture broadcast possibility. :-))

73' Jan, PA3GJX




Op ma 6 jul. 2020 19:16 schreef Scott Howard <notifications@github.com>:

**Comment by maqifrnswa** [comment]
That error coming from too large of a picture makes sense. The maximum dimension (either width or height) of an image is 4096 px. That should be in the program somewhere. Theoretically, image size doesn't matter, but we just had to put a limit somewhere. I suggest starting off with SDTV quality images (around 320x240) just because the image comes in to focus faster.

As to your question:
> On my Windows 10 the gui does not save the station information, can it be i have done something wrong installing?

Do you mean saving your callsign after you close the program? Right now, the program doesn't do that. Every time you open it, it starts off with "NOCALL". I haven't implemented saving settings yet, but that's something that can be done.

Thanks again, it's pretty cool to have someone trying this out in the Netherlands! 73

**Comment by maqifrnswa** [comment]
closing this because Windows binaries have been released and used by several people. Also, as of yesterday, Mac binaries are available for MacOS 10.14 and 10.15! The next release will contain Windows and Mac binaries.
